File: admin/app/routers/mirror_process.py
from fastapi import APIRouter, Request

router = APIRouter()


@router.post("/api/mirror-node/start")
def post_mirror_node_start(request: Request):
    return request.app.state.mirror_process.start()


@router.post("/api/mirror-node/stop")
def post_mirror_node_stop(request: Request):
    return request.app.state.mirror_process.stop()


@router.get("/api/mirror-node/status")
def get_mirror_node_status(request: Request):
    return request.app.state.mirror_process.status()

File: admin/frontend/src/api/mirrorProcess.ts
import { apiFetch } from "./client";

export interface MirrorProcessStatus {
  running: boolean;
  pid: number | null;
}

export function startMirrorProcess(): Promise<MirrorProcessStatus> {
  return apiFetch<MirrorProcessStatus>("/api/mirror-node/start", { method: "POST" });
}

export function stopMirrorProcess(): Promise<MirrorProcessStatus> {
  return apiFetch<MirrorProcessStatus>("/api/mirror-node/stop", { method: "POST" });
}

export function getMirrorProcessStatus(): Promise<MirrorProcessStatus> {
  return apiFetch<MirrorProcessStatus>("/api/mirror-node/status");
}
